docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
+
+The following diagram shows the sequence diagram of the addition of an item.
+
+The user starts by typing an add command. The example used in the diagram above is the addition of an item with the name `Paper Cup`, quantity of `25` and description of `100ml paper cups`. The full command is `add n/Paper Cup q/25 d/100ml paper cups`.
+
+1. The `run()` method within `InvMgr` calls the static method `parse()` in the `Parser` class, providing the entire string of input entered by the user.
+2. Within `parse()`, the string is identified to start with the word `add`, and executes the code within the case. The case attempts to create an item using the string by self-invoking `createItem()`.
+3. `createItem()` extracts the relevant arguments from the input string and generates a new `Item` which is returned to the `parse()` case block.
+4. `parse()` uses the `Item` to generate a new `AddCommand` which is returned to the `run()` method.
+5. The `run()` method calls on the `execute()` function in the `AddCommand` which will add the generated item to the `ItemList` using its `addItem()` method.
+6. `AddCommand` will converse with `Ui` to show a message that the item has been added. In this case, the item to add will be printed as the name of the item, followed by " has been added!".
+
+### Delete Command
+![DeleteCommandSequenceDiagram](img/DeleteCommandSequenceDiagram.png)
+
+The following diagram shows the sequence diagram of the addition of an item.
+
+The user starts by typing an add command. The example used in the diagram above is the addition of an item with the index `1`, based on the list when the user types the `list` command.
+
+1. The `run()` method within `InvMgr` calls the static method `parse()` in the `Parser` class, providing the entire string of input entered by the user.
+2. Within `parse()`, the string is identified to start with the word `delete`, and executes the code within the case. The case finds the index of the item by splitting the string and indexing it.
+3. `parse()` generates a new `AddCommand` using the index which is returned to the `run()` method.
+4. The `run()` method calls on the `execute()` function in the `DeleteCommand` which will delete the item with that index from the `ItemList` using its `removeItem()` method.
+5. `DeleteCommand` will converse with `Ui` to show a message that the item has been removed. In this case, the item to add will be printed as the name of the item, followed by " has been deleted.".
+
+### List Command
+![ListCommandSequenceDiagram](img/ListCommandSequenceDiagram.png)
+
+The following diagram shows the sequence diagram of the listing of items in `itemList`.
+
+The user starts by typing a list command.
+
+1. `InvMgr` calls `parse("list")` method in `Parser` class, which returns a ListCommand object.
+2. `InvMgr` calls `execute(itemList, ui)` method in `ListCommand` object.
+3. `ListCommand` loops through every `Item` in `itemList` and prints them line by line
+   and numbers them.
```

Comment:
![image](https://user-images.githubusercontent.com/69455499/160964986-c78cea65-4825-450e-b147-f44cea5b9f24.png)
Consider moving the end of lifeline cross out of the object box and at the end of the line.